### ISSUE TITLE
Detox globals no longer available in utility functions (outside of tests)

### ISSUE THREAD

**Issue by abury** [wix/Detox#1353]
**Describe the bug**
Hey Team

We've recently upgraded to RN 0.59 from a much earlier version. When we first started we wrote a number of utility functions that were stored in a separate utility file. These helped us write more concise specs and provided an enhanced debugging experience:

spec:

```javascript
import { APPNAME } from '../aliases';

describe('Do a thing', () => {
  describe('Show list of things', () => {
    beforeEach(async () => {
      await device.reloadReactNative();
    });

    it('should be able to do thing 1', async () => {
      await APPNAME.withText('$0').expect.toBeVisible();
    });
  });
});
```

aliases/index.js
````javascript
const withText = text => {
  return element(by.text(text));
};
````

This was working very well until updating RN. Now we see the following error:
````
    ReferenceError: element is not defined

       8 | 
       9 |     const withLabel = text => {
    > 10 |       return element(by.label(text));
         |       ^
````

It seems that element (and other detox globals) are no longer accessible outside of an 'it' statement. I thought at first it might be a jest issue, but I've downgraded jest to the version we were previously using and we're still experiencing the issue.

Have there been any behind the scenes changes that stop detox globals from being accessible outside of a test? Any insight would be really appreciated.

Thanks 👍

**To Reproduce**

- [x ] I have tested this issue on the latest Detox release and it still reproduces

**Expected behavior**
Detox globals, such as `element` are accessible outside tests in external utility functions

**Screenshots**
N/A

**Environment (please complete the following information):**
 - Detox: 12.6.1
 - React Native: 0.59.5
 - Node: 10.0.0
 - Device: N/A
 - Xcode: 10.2.1
 - iOS: 12.2
 - macOS: 10.14.4

**Device and Verbose Detox Logs**
````
detox test --configuration ios.sim.debug -f e2e/doThings/thing.spec.js --loglevel trace
detox[75892] WARN:  [deprecation.js] Beware: -f, --file will be removed in the next version of Detox.
detox[75892] WARN:  [deprecation.js] See the migration guide:
https://github.com/wix/Detox/blob/master/docs/Guide.Migration.md#migrating-from-detox-120x-to-121x 
detox[75892] INFO:  [test.js] configuration="ios.sim.debug" loglevel="trace" artifactsLocation="artifacts/ios.sim.debug.2019-05-02 05-26-42Z" recordLogs="none" takeScreenshots="manual" recordVideos="none" recordPerformance="none" node_modules/.bin/jest --config=e2e/config.json --maxWorkers=1 '--testNamePattern=^((?!:android:).)*$' "e2e/doThings/thing.spec.js"
 FAIL  e2e/doThings/thing.spec.js
  ● Test suite failed to run

    ReferenceError: element is not defined

       5 | };
       6 | const withId = id => {
    >  7 |   return ElementProxy(element(by.id(id)), 'id', id);
         |                              ^
       8 | };
       9 | const withText = text => {
      10 |   return ElementProxy(element(by.text(text)), 'text', text);

      at withId (e2e/aliases/index.js:7:30)
      at Object.<anonymous> (e2e/aliases/index.js:20:9)
      at Object.<anonymous> (e2e/doThings/thing.spec.js:1:222)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        1.352s
Ran all test suites matching /e2e\/doThings\/thing.spec.js/i with tests matching "^((?!:android:).)*$".
detox[75892] ERROR: [cli.js] Error: Command failed: node_modules/.bin/jest --config=e2e/config.json --maxWorkers=1 '--testNamePattern=^((?!:android:).)*$' "e2e/doThings/thing.spec.js"
````

**Comment by rotemmiz** [comment]
Hey @abury,
Nothing has changed in Detox in that regard. Did you also upgrade jest or node? I would investigate that direction as well.
Anyway, you can always disable global exports and require detox API manually.

**Comment by abury** [comment]
Hey @rotemmiz, thanks for the reply.
We did upgrade jest, but I've tried downgrading it and unfortunately, it didn't do anything. I'll try requiring the detox API manually 👍

**Comment by abury** [comment]
Unfortunately, that's not working either. Whilst `expect` and `element` is defined, `by` is undefined when importing it from detox. I followed the instructions from https://github.com/wix/Detox/blob/a954203828f2a466d77a148050ac133b8765df96/docs/APIRef.DetoxObjectAPI.md

I even tried importing `by` directly from ios expect:
`import { by } from '../../node_modules/detox/src/ios/expect';`

And received the following error
`TypeError: Cannot read property 'apply' of undefined`

which is confusing as we're not calling anything called `apply` anywhere.
If you've got any insights as to why this might be occurring, it would be amazing.

Thanks,

Aron

**Comment by rotemmiz** [comment]
You're holding it wrong 😉
https://github.com/wix/Detox/blob/master/docs/APIRef.DetoxObjectAPI.md#explicit-imports-during-initialization

**Comment by abury** [comment]
@rotemmiz No, it was being imported properly and I followed the instructions in that link. The import direct from node_modules was just me trying to debug things.

It turned out to be a timing issue. We used to call `element` before `detox.init()` was called (thanks to an import and lazy instantiation), and it (somehow) used to work without an issue. Since updating react native, it has (rightly) broken.

It's one of those issues where it should never have worked in the first place 😂. We've re-arranged things and it's all working now.
